pkg/controller/vitesscluster/reconcile_vtctld.go

```diff
@@ -177,7 +177,16 @@ func (r *ReconcileVitessCluster) vtctldSpecs(vt *planetscalev2.VitessCluster, pa
 			Annotations:       vt.Spec.VitessDashboard.Annotations,
 			ExtraLabels:       vt.Spec.VitessDashboard.ExtraLabels,
 			Tolerations:       vt.Spec.VitessDashboard.Tolerations,
-		})
+			BackupEngine:      vt.Spec.Backup.Engine,
+		}
+		// We don't know which location to use for vtctld.
+		// If there are more than one locations configured, we can only choose one.
+		// Here we opt to take the first one.
+		if len(vt.Spec.Backup.Locations) != 0 {
+			spec.BackupLocation = &vt.Spec.Backup.Locations[0]
+		}
+		specs = append(specs, spec)
```

Comment:
This here is worth noting. A user can setup multiple backup locations and use different ones in different tablet pools. However, we can only configure the vtctld with the flags to support one implementation. For now I have chosen to take the first one.

Reply:
This should be fine. Even though operator's spec allows this, it is not supported by Vitess itself.